### ISSUE TITLE
Empty response body cannot be dereferenced in :completed handler

### ISSUE THREAD

**Issue by pbostrom** [cch1/http.async.client#69]
For example, the following async request will return an empty response body, which will cause the specified `:completed` handler to hang forever:
``` clojure
(execute-request client (prepare-request :get "http://localhost:8123/empty")
                 :completed (fn [response] @(:body response)))
```
Under the netty async http provider, this has the additional effect of tying up the `run()` loop in `AbstractNioSelector`:
https://github.com/netty/netty/blob/netty-3.10.4.Final/src/main/java/org/jboss/netty/channel/socket/nio/AbstractNioSelector.java#L207
which will ultimately cause any calls to close the async client to hang forever.

**Comment by cch1** [comment]
Looking forward to a PR that will deliver an empty body before the `completed` callback is invoked.

**Comment by pbostrom** [comment]
Fixed in 5f8768a

**Comment by cch1** [comment]
Thanks Paul!
